### PR TITLE
feat: add `options` field and improve formatting of errors

### PR DESCRIPTION
**Pull request by picunada**
### 🔗 Linked issue

Issue #265 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

As fetch options extending `RequestInit` object, it will be more convenient to return `FetchOptions` object in  `options` field of `FetchError` object.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/error.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import type { FetchRequest, FetchResponse } from "./fetch";
+import type { FetchOptions, FetchRequest, FetchResponse } from "./fetch";
 
 export class FetchError<T = any> extends Error {
   name = "FetchError";
-  request?: FetchRequest;
+  request?: FetchRequest | FetchOptions;
```
Comment (pi0) [inline src/error.ts:5]: I think we can have an additional `options: FetchOptions` field

File: src/error.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export function createFetchError<T = any>(
   if (request && response) {
     message = `${message} (${response.status} ${
       response.statusText
-    } (${request.toString()}))`;
+    } (${JSON.stringify(request)}))`;
```
Comment (pi0) [inline src/error.ts:26]: Here, then we can have conditions to add request + additional info such as method from options

File: src/fetch.ts [hunk 3]
```diff
@@ -119,7 +119,7 @@ export function createFetch(globalOptions: CreateFetchOptions): $Fetch {
 
     // Throw normalized error
     const error = createFetchError(
-      context.request,
+      typeof context.request === "string" ? context.options : context.request,
```
Comment (pi0) [inline src/fetch.ts:122]: We shall make both available

Comment (picunada) [reply src/fetch.ts:122]: Ok, I will make changes

File: src/error.ts (hunk 1 above)
Comment (picunada) [reply src/error.ts:5]: > I think we can have an additional `options: FetchOptions` field

Yes, that's more correct.

**Comment by picunada** [comment]
@pi0 
Done refactoring code.

**Comment by picunada** [comment]
Why is CodeCov failing in CI?
```bash
[2023-08-21T22:55:56.262Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

**Comment by picunada** [comment]
Sorry for amount of commits, I was trying to make codecov working.
It was failing to upload commits(it can be checked in previous gh actions for this PR).
Should we add anything to docs?

**Comment by pi0** [comment]
Thanks for recent changes! I might need little bit more time to properly review this on. Planning for next minor release 👍🏼

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**